In [11]:
import pandas as pd
import numpy as np
import matplotlib as plt
import pprint
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [8]:
df = pd.read_stata('hprice1.dta')
df

price      assess  bdrms  lotsize  sqrft  colonial    lprice   lassess  \
0   300.0  349.100006      4   6126.0   2438         1  5.703783  5.855359   
1   370.0  351.500000      3   9903.0   2076         1  5.913503  5.862210   
2   191.0  217.699997      3   5200.0   1374         0  5.252274  5.383118   
3   195.0  231.800003      3   4600.0   1448         1  5.273000  5.445875   
4   373.0  319.100006      4   6095.0   2514         1  5.921578  5.765504   
..    ...         ...    ...      ...    ...       ...       ...       ...   
83  295.0  318.299988      3   6056.0   1837         1  5.686975  5.762994   
84  236.0  259.399994      3   5828.0   1715         0  5.463832  5.558371   
85  202.5  258.100006      3   6341.0   1574         0  5.310740  5.553347   
86  219.0  232.000000      2   6362.0   1185         0  5.389072  5.446737   
87  242.0  252.000000      4   4950.0   1774         1  5.488938  5.529429   

    llotsize    lsqrft  
0   8.720297  7.798934  
1   9.200593  7.638198  
2   8.556414  7.225482  
3   8.433811  7.277938  
4   8.715224  7.829630  
..       ...       ...  
83  8.708805  7.515889  
84  8.670429  7.447168  
85  8.754792  7.361375  
86  8.758098  7.077498  
87  8.507143  7.480992  

[88 rows x 10 columns]

In [14]:
pprint.pprint(pd.io.stata.StataReader('hprice1.dta').variable_labels())

{'assess': 'assessed value, $1000s',
 'bdrms': 'number of bdrms',
 'colonial': '=1 if home is colonial style',
 'lassess': 'log(assess',
 'llotsize': 'log(lotsize)',
 'lotsize': 'size of lot in square feet',
 'lprice': 'log(price)',
 'lsqrft': 'log(sqrft)',
 'price': 'house price, $1000s',
 'sqrft': 'size of house in square feet'}


In [21]:
model = smf.ols(formula = 'price ~ sqrft + bdrms', data = df)
results = model.fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                  price   R-squared:                       0.632
Model:                            OLS   Adj. R-squared:                  0.623
Method:                 Least Squares   F-statistic:                     72.96
Date:                Tue, 02 Sep 2025   Prob (F-statistic):           3.57e-19
Time:                        15:21:18   Log-Likelihood:                -488.00
No. Observations:                  88   AIC:                             982.0
Df Residuals:                      85   BIC:                             989.4
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    -19.3150     31.047     -0.622      0.5

i. regression model price = -19.315 + 0.1284sqrt + 15.1982bdrms
ii. A house with one exra bedroom will increase the price of the house by 15.19 thousand dollars
iii. The estimated price increase is 33.166 (15.19 + (140*.1284))
iv. 63.2 percent of the variation in is explained by number of bedrooms and square footage
v. first house estimate sqrt = 2438 and bdrms = 4 Estimated to be 60.76 + 313.04 - 19.31 = 354.49
vi. residual = -54.49 and this suggets that that the buyer underpayed for the house